java/test/src/main/java/io/ray/test/ReferenceCountingTest.java

```diff
@@ -0,0 +1,275 @@
+package io.ray.test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import io.ray.api.ActorHandle;
+import io.ray.api.ObjectRef;
+import io.ray.api.Ray;
+import io.ray.api.id.ObjectId;
+import io.ray.runtime.object.NativeObjectStore;
+import io.ray.runtime.object.ObjectRefImpl;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Test
+public class ReferenceCountingTest extends BaseTest {
+  @BeforeClass
+  public void setUp() {
+    System.setProperty("ray.object-store.size", "100 MB");
+  }
+
+  @AfterClass
+  public void tearDown() {
+    System.clearProperty("ray.object-store.size");
+  }
+
+  @BeforeMethod
+  public void setUpCase() {
+    TestUtils.skipTestUnderSingleProcess();
+  }
+
+  /**
+   * Because we can't explicitly GC an Java object. We use this helper method to manually remove an
+   * local reference.
+   */
+  private static void del(ObjectRef<?> obj) {
+    ((ObjectRefImpl<?>) obj).removeLocalReference();
```

Comment:
Can we instead let the object go out of scope and manually call gc()?

Reply:
Unfortunately, we can't do this in Java. See https://stackoverflow.com/questions/1481178/how-to-force-garbage-collection-in-java.